fishtank/src/backend/docker.ts

```diff
@@ -86,6 +86,24 @@ export class Docker {
     await this.cmd(runArgs, {})
   }
 
+  async remove(
+    containers: string[],
+    options?: { force?: boolean; volumes?: boolean },
+  ): Promise<void> {
+    if (containers.length === 0) {
+      return
+    }
+    const rmArgs = ['rm']
+    if (options?.force) {
+      rmArgs.push('--force')
+    }
+    if (options?.volumes) {
+      rmArgs.push('--volumes')
+    }
+    rmArgs.push(...containers)
+    await this.cmd(rmArgs, {})
```

Comment:
How do we know if container is removed successfully?

Reply:
If Docker does not throw any error, then arguably all the containers have been removed. If Docker does throw an error, then a TypeScript exception will pop up and in that case it's possible that some containers may still be running.

Reply:
One edge case I can think of is if the container name is not found? will the child process throw an error? I know on the docker side it will have stderr with not found error message.

Reply:
With `--force`, no error is thrown if the container does not exist. Without `--force`, an error is thrown. This can be easily verified with an example:

```sh
$ docker rm does-not-exist
Error response from daemon: No such container: does-not-exist
$ echo $?
1

$ docker rm -f does-not-exist
Error response from daemon: No such container: does-not-exist
$ echo $?
0
```

So, depending on whether you pass `force` or not, you may or may not get an error. Does this answer the question?

Reply:
Okay, I read more about the child process from Node.js website. `execFile` will throw error if any exit code other than 0. Looks like not found is already handled by your code :+1: